File: android/app/src/main/java/org/swalsh/life/SettingsOpener.java
package org.swalsh.life;

import android.content.Intent;
import android.net.Uri;
import android.provider.Settings;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class SettingsOpener extends ReactContextBaseJavaModule {
    private ReactContext context;

    public SettingsOpener(ReactApplicationContext context) {
        super(context);
        this.context = context;
    }

    @Override
    public String getName() {
        return "SettingsOpener";
    }

    @ReactMethod
    public void openSettings() {
        Intent intent = new Intent();
        Uri uri = Uri.fromParts("package", context.getPackageName(), null);

        intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
        intent.setData(uri);

        context.startActivity(intent);
    }
}

File: android/app/src/main/kotlin/org/swalsh/life/SettingsOpener.kt
package org.swalsh.life

import android.content.Intent
import android.net.Uri
import android.provider.Settings

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class SettingsOpener(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
    private val context: ReactContext

    init {
        this.context = context
    }

    override fun getName(): String {
        return "SettingsOpener"
    }

    @ReactMethod
    fun openSettings() {
        val intent = Intent()
        val uri = Uri.fromParts("package", context.packageName, null)

        intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
        intent.data = uri

        context.startActivity(intent)
    }
}
